### PR TITLE
various: change `libcython` to `cython`

### PR DESCRIPTION
**Pull request by p-linnane**
Signed-off-by: Patrick Linnane <patrick@linnane.io>

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----
Needed for https://github.com/Homebrew/homebrew-core/pull/166330.

### CODE REVIEW

**Comment by p-linnane** [comment]
@chenrui333 Some of these cannot be fetch/built. We just need to do a syntax-change to get the other PR shipped.

**Comment by Bo98** [comment]
I'd really rather this was tested and the things that can't build already be split to separate PRs.

**Comment by p-linnane** [comment]
That's fine. Will split them out later on. `pytouhou` in particular will never work, because the source can't be fetched anymore.

**Comment by Bo98** [comment]
Ok, though it's probably worth thinking ahead of strategy for Python 3.13 migration. Do we plan to migrate the `cython` dependent tree together?

File: Formula/o/onlykey-agent.rb [hunk 1]
```diff
@@ -221,7 +221,7 @@ def install
     # add path configuration file to find cython
     site_packages = Language::Python.site_packages(python3)
     pth_contents = <<~EOS
-      import site; site.addsitedir('#{Formula["libcython"].opt_libexec/site_packages}')
+      import site; site.addsitedir('#{Formula["cython"].opt_libexec/site_packages}')
     EOS
     (libexec/site_packages/"homebrew-onlykey-agent.pth").write pth_contents
   end
```
Comment (Bo98) [inline Formula/o/onlykey-agent.rb:227]: This one will require a proper rebottle (or revision bump).

**Comment by p-linnane** [comment]
I am not sure how we should handle it long term. Just trying to unblock @cho-m's PR here.

**Comment by cho-m** [comment]
`cython` and `libcython` have remained `keg_only` and require pth hacks so may just move most runtime usage to resources. I don't even know if we were using correct API after the `0.29.x --> 3.x` release

Also, with build isolation (difficult to disable these days due to `distutils` removal), our `cython` is rarely used as build dependency. Only time I think we need it is when a formula code has to be re-`cythonize`d (e.g. new Python release).

**Comment by p-linnane** [comment]
https://github.com/Homebrew/homebrew-core/pull/166558
https://github.com/Homebrew/homebrew-core/pull/166559
https://github.com/Homebrew/homebrew-core/pull/166560
https://github.com/Homebrew/homebrew-core/pull/166561
https://github.com/Homebrew/homebrew-core/pull/166562

**Pull request closed without merging**